pkg/action/install_test.go

```diff
@@ -79,62 +58,68 @@ func TestInstallAllSharedDeps(t *testing.T) {
 		addRelStub     bool
 		optionalDeps   optionalDepsStrategy
 		wantNSFromFlag string
+		numRels        int
 	}{
 		{
-			name:      "chart has no shared-deps",
-			chart:     buildChart(withHypperAnnotations()),
-			golden:    "output/install-no-shared-deps.txt",
-			wantDebug: true,
-		},
-		{
-			name:      "chart metadata has malformed yaml",
-			chart:     buildChart(withMalformedSharedDeps()),
-			golden:    "output/install-malformed-shared-deps.txt",
-			wantError: true,
-			error:     "yaml: line 2: mapping values are not allowed in this context",
-		},
-		{
-			name:   "dependencies get correctly installed",
-			chart:  buildChart(withHypperAnnotations(), withSharedDeps()),
-			golden: "output/install-correctly-shared-deps.txt",
-		},
-		{
-			name:   "dependencies without annotations get correctly installed",
-			chart:  buildChart(withHypperAnnotations(), withSharedDepsWithoutAnnotations()),
-			golden: "output/install-shared-deps-without-annotations.txt",
-		},
-		{
-			name:           "dependencies with NamespaceFromFlag get correctly installed",
-			chart:          buildChart(withHypperAnnotations(), withSharedDeps()),
-			golden:         "output/install-shared-deps-with-ns-from-flag.txt",
-			wantNSFromFlag: "ns-from-flag",
-		},
-		{
-			name:       "dependencies are already installed",
-			chart:      buildChart(withHypperAnnotations(), withSharedDeps()),
-			golden:     "output/install-shared-dep-installed.txt",
+			name:       "chart has no shared-deps",
+			chart:      buildChart(withHypperAnnotations()),
+			golden:     "output/install-no-shared-deps.txt",
 			addRelStub: true,
+			numRels:    1,
 		},
-		{
-			name:       "dependencies are already installed in out-of-range ver",
-			chart:      buildChart(withHypperAnnotations(), withOutOfRangeSharedDeps()),
-			golden:     "output/install-shared-dep-installed-out-of-range.txt",
-			addRelStub: true,
-			wantError:  true,
-			error:      "Shared dep version out of range; 0.1.0 is not equal to 1.1.0",
-		},
-		{
-			name:         "optional dependencies get correctly installed",
-			chart:        buildChart(withHypperAnnotations(), withOptionalSharedDeps()),
-			golden:       "output/install-correctly-optional-shared-deps.txt",
-			optionalDeps: OptionalDepsAll,
-		},
-		{
-			name:         "optional dependencies get correctly skipped",
-			chart:        buildChart(withHypperAnnotations(), withOptionalSharedDeps()),
-			golden:       "output/skip-optional-shared-deps.txt",
-			optionalDeps: OptionalDepsNone,
-		},
+		// {
+		// 	name:      "chart metadata has malformed yaml",
+		// 	chart:     buildChart(withMalformedSharedDeps()),
+		// 	golden:    "output/install-malformed-shared-deps.txt",
+		// 	error:     "yaml: line 2: mapping values are not allowed in this context",
+		// 	wantError: true,
+		// 	numRels:   0,
+		// },
```

Comment:
Will all these tests be set to work again?

Reply:
Yes, these are the non-working ones we were talking about today.

Reply:
Re-enabled `pkg/action/install_test.go` in https://github.com/rancher-sandbox/hypper/pull/176/commits/40733b95d8447d06a061d96e4ddd6a023d5df51c.

Needed refactors and bug fixes for them to pass: 
https://github.com/rancher-sandbox/hypper/pull/176/commits/f9c672ec8fb662ad80515b1aa2d61a83a2c3b8f3
https://github.com/rancher-sandbox/hypper/pull/176/commits/1d790beb9a9554d32174ac8f5cf87d8ad5f09939
https://github.com/rancher-sandbox/hypper/pull/176/commits/b242bb5500c3e6842be7ce5c671c4a00685406d0
https://github.com/rancher-sandbox/hypper/pull/176/commits/647d5db1baeaa2fcd6e2f4091ed4c656394ee161
https://github.com/rancher-sandbox/hypper/pull/176/commits/b2b9f950d7af303f141893564ca89b26448b4edf
https://github.com/rancher-sandbox/hypper/pull/176/commits/43d8c30966bbcd869ab12536705a48a7ed968015
https://github.com/rancher-sandbox/hypper/pull/176/commits/be0ba44489fe2a2bba00197317a51cb92b9b40a5
https://github.com/rancher-sandbox/hypper/pull/176/commits/4f0ed3e9e11546be49b6cfef7b8fd9c919ad5fde
https://github.com/rancher-sandbox/hypper/pull/176/commits/9d5fd6572af4b7b4a47918ae91e71bbc735805da